tools/validators/instance_validator/instance_parser.py

```diff
@@ -100,26 +100,36 @@ def parse_yaml(filename):
     filename: filepath location of the YAML file
 
   Returns:
-    Returns the parsed YAML data in a stricyaml-provided
-    datastructure which is similar to a Python dictionary.
+    Returns a tuple with the parsed YAML data in a stricyaml-provided
+    datastructure which is similar to a Python dictionary, and a
+    possible exception.
   """
-  yaml = _load_yaml_with_schema(filename, _SCHEMA)
+  yaml, err = _load_yaml_with_schema(filename, _SCHEMA)
 
   if yaml is None:
-    return None
+    return (None, err)
 
   top_name = yaml.keys()[0]
 
   if _TRANSLATION in yaml[top_name].keys():
     translation = yaml[top_name][_TRANSLATION]
     translation.revalidate(_TRANSLATION_SCHEMA)
 
-    # TODO can this be automatically verified based on ontology?
     # if translation is not UDMI compliant
-    if translation.data != _COMPLIANT:
+    if isinstance(translation.data, str):
+      if translation.data != _COMPLIANT:
+        return (None, Exception('Translation compliance improperly defined'))
+    else:
       translation_keys = translation.keys()
 
       for key in translation_keys:
-        translation[key].revalidate(_TRANSLATION_DATA_SCHEMA)
-
-  return yaml
+        try:
+          translation[key].revalidate(_TRANSLATION_DATA_SCHEMA)
+        except (ruamel.yaml.parser.ParserError,
+                ruamel.yaml.scanner.ScannerError,
+                syaml.exceptions.YAMLValidationError,
+                syaml.exceptions.DuplicateKeysDisallowed,
+                syaml.exceptions.InconsistentIndentationDisallowed) as e:
+          return (None, e)
```

Comment:
returning the exception is enough why the need to return None as well?

Reply:
Because the function is now defined to return a tuple, so this avoids the errors when unpacking the return values later